File: src/main/java/model/equipment/weapon/Weapon.java
package model.equipment.weapon;

import model.equipment.Equipment;
import model.support.allEnums.DmgType;
import model.support.allEnums.EquipmentType;
import model.support.allEnums.WeaponType;

import java.util.HashMap;
import java.util.Map;

import static model.support.Constants.FIVE;
import static model.support.Constants.TWO;

public abstract class Weapon extends Equipment {
    protected WeaponType weaponType;
    protected boolean isOneHanded;

    protected Map<DmgType, Double> damage = new HashMap<>();

    {
        damage.put(DmgType.BLUNT, 0.0);
        damage.put(DmgType.CUT, 0.0);
        damage.put(DmgType.SLASH, 0.0);
        damage.put(DmgType.PIERCING, 0.0);
        damage.put(DmgType.CRUSH, 0.0);
    }

    public Weapon(WeaponType weaponType, double weight, int quality) {
        super(EquipmentType.WEAPON);
        this.weaponType = weaponType;
        this.weight = weight;
        this.quality = quality;
        this.price = quality / FIVE - weight * TWO;
    }

    public double getPrice() {
        return price;
    }

    public double getQuality() {
        return quality;
    }

    public double getWeight() {
        return weight;
    }

    public WeaponType getWeaponType() {
        return weaponType;
    }

    public boolean isOneHanded() {
        return isOneHanded;
    }

    public Map<DmgType, Double> getDamage() {
        return damage;
    }
}

File: src/main/java/model/equipment/weapon/Weapon.kt
package model.equipment.weapon

import model.equipment.Equipment
import model.support.Constants.FIVE
import model.support.Constants.TWO
import model.support.enums.EquipmentType.WEAPON
import model.support.enums.Skill
import model.support.enums.Specification
import model.support.enums.WeaponType

abstract class Weapon(
        val weaponType: WeaponType,
        val skillType: Skill,
        weight: Double,
        quality: Int,
        val isOneHanded: Boolean
) : Equipment(
        WEAPON,
        weight,
        quality,
        quality / FIVE - weight * TWO
) {
    var damage = mapOf<Specification, Double>()
        protected set
}
